other-tests/character-class.js

```diff
@@ -0,0 +1,25 @@
+// Copyright 2017 Mathias Bynens. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+author: Mathias Bynens
+description: >
+  Unicode property escapes must be supported in character classes.
+esid: sec-static-semantics-unicodematchproperty-p
+features: [regexp-unicode-property-escapes]
+---*/
+
+/[\p{Hex}]/u;
+assert(
+  /[\p{Hex}-\uFFFF]/u.test('-'),
+  'property escape in character class should not be parsed as the start of a range'
+);
+assert.throws(SyntaxError, () => /[\p{}]/u);
```

Comment:
you can not even catch them in a `eval('/\uGGGG/')`, I'm assuming the same will happen for these.

I forgot to paste the link for negative tests yesterday: https://github.com/tc39/test262/blob/master/INTERPRETING.md#negative